src/test/java/com/puppycrawl/tools/checkstyle/grammar/comments/AllBlockCommentsTest.java

```diff
@@ -48,14 +44,15 @@ protected String getPackageLocation() {
 
     @Test
     public void testAllBlockComments() throws Exception {
```

Comment:
Please investigate a bit and leave a comment above this test method (or class) that describes why we need such a strange test.

Reply:
@nrmancuso Honestly idk I found it weird too. just updated it to be an example for the issue. but I will investigate

Reply:
https://github.com/checkstyle/checkstyle/pull/14537

Reply:
@mahfouz72 if CI is green in https://github.com/checkstyle/checkstyle/pull/14537, please just do as I have there and delete this class completely.

Reply:
@nrmancuso should I close this PR? or just update it after the removal

Reply:
Please do:
1. Have exactly one commit with test class removal
2. Make your commit message like `Pull #14518: Remove AllBlockCommentTest` to have a quick reference back to our conversation here
3. Edit PR title to match

Reply:
@nrmancuso done

Reply:
Leaving this unresolved so that next reviewer can see why we removed this.

Reply:
This test just test our ability to detect comments at any possible place.  It might not be required for coverage but it helps to control parsing

Reply:
Input file that is full of block multi line comments https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/grammar/comments/InputFullOfBlockComments.java

We just need to change
```
/*0*//*1*/package/*2*/ com/*3*/./*4*/puppycrawl/*5*/./*6*/tools/*7*/./*8*/checkstyle.grammar/*9*/./*10*/comments/*11*/;/*12*/
/*13*/

```
To more verbose:
```
/*0*//*1*/package/*2*/ com/*3*/./*4*/puppycrawl/*5*/./*6*/tools/*7*/./*8*/checkstyle.grammar/*9*/./*10*/comments/*11*/;/*12*/
// 12 violations above:
//.           '0'
//.           '1'

```

Reply:
> This test just test our ability to detect comments at any possible place. It might not be required for coverage but it helps to control parsing

Ok, then let’s create a test for the AST with comments generated by this file.

Reply:
yes, good idea, this is exactly what we wanted many years ago, but we did not have this concept at all.

Reply:
@mahfouz72 , please move this test to AST tests
Example: https://github.com/checkstyle/checkstyle/blob/8531fec143058ddd92fab14839f2e9389ac6cf5c/src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java#L73-L76

Reply:
@romani done I have moved singlelineCommentTest also to AST test. CI will be green after merging https://github.com/checkstyle/checkstyle/pull/14567